Qurator docs (#4159)
quiltdata/quilt: docs/Catalog/Qurator.md

# Qurator Omni Beta

`Qurator Omni` is an always-available AI assistant embedded directly into the

> Quilt web catalog. It allows users to interact with Quilt packages and search

Reviewer: there's no "interaction with quilt packages" currently implemented (aside from searching them and accessing file/entry contents)

Author: "S3 buckets"

functionality through natural language. Qurator Omni leverages advanced models
like Claude, integrated via Amazon Bedrock, enabling users to query, retrieve, and

> summarize data instead of having click through the GUI.

Author: ```suggestion
summarize data instead of having to click through the GUI.
```

> ## What is Qurator Omni?

Author: removed headings

- **Instant Summaries**: Quickly digest scientific papers, datasets, or reports
  without reading everything.
- **Fine-Grained Permissions with RAG**: Query only the data you're authorized
  to access, ensuring compliance with strict organizational policies
- **Secure Cloud Environment**: Work within your private AWS cloud, ensuring
  data stays secure while using state-of-the-art AI models.

## Getting Started

To enable Qurator Omni:

1. **Opt-In to Qurator**:  
   - Install Release 1.55 or later of the Quilt Platform CloudFormation template.
   - Set the `qurator` parameter to `true` in the CloudFormation template to
     enable the Qurator chatbot.

2. **Configure Claude Model**:  
   - Login to the Amazon Bedrock console.
   - Ensure that the Claude 3.5 Sonnet model is available in the same region as
     your Quilt deployment. Check [Model support by AWS
     Region](https://docs.aws.amazon.com/bedrock/latest/userguide/models-regions.html)
     for details.
   - Enable the model by configuring it within your Bedrock environment.
   - Carefully monitor the model’s cost implications. The Claude model is
     charged based on usage, so ensure that you have the necessary budget

>      allocated. Initial estimates are roughly a penny per page for complex documents,

Reviewer: ```suggestion
     allocated. Initial estimates are roughly a penny per page for complex documents.
```
or were you meaning to continue the sentence?

Reviewer: also, "complex" or not -- doesn't actually matter

Author: I concede "complex" is somewhat grauitous, but AI does do a better job of interpreting complex items like tables.

3. **Start Using Qurator**:  
   - Once activated, the Qurator chatbot will appear in the Quilt web catalog
     interface.
   - Click the Qurator icon on the
     bottom right of the screen to open the chat interface. ![qurator icon](../imgs/qurator-icon.png)
   - You can begin by typing questions into the chat interface. For example,
     queries like _“What are the key findings on small molecule delivery?”_ will

>      prompt Qurator to retrieve relevant packages and present a summarized

Reviewer: ```suggestion
     prompt Qurator to search for relevant data and present a summarized
```

     overview.

### Example Use Cases

- **Search**: _“What are the latest papers on melanoma?”_  

>   Qurator will search through the Quilt packages using Elastic Search and

Reviewer: ```suggestion
  Qurator will search through the Quilt catalog using Elastic Search and
```

  retrieve the most relevant data.
  
- **Summarize**: _“Summarize the key points of this BRCA1 research.”_  
  After selecting a specific document, Qurator will generate a clear, useful
  summary of the paper.

- **Quick Scan**: _“List some of the authors doing breast cancer research?”_  

>   Qurator Omni will list key authors and their contributions based on Quilt’s

Reviewer: not sure about "key" here

Author: removed

  indexed datasets.

### Key Benefits

- **Enhanced Productivity**: Eliminate the need for manual search navigation,
  enabling faster access to critical information.
- **Improved Insights**: Gain deeper insights from large datasets with automatic
  summaries.

> - **Streamlined Collaboration**: Easily share summarized findings and insights

Reviewer: i'm not quite sure what exactly this is supposed to mean

Author: Good point. Trying this "Leveraging AI chat to provide background and
  context when working across disciplines."
